### PR TITLE
chore: Add NFT Service tests

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR does the following:
- Adds tests for the NFT Service.
- Rewrites the fetch method, fixing a bug where the accounts were not computed correctly.

### CODE REVIEW

File: webapp/src/modules/vendor/decentraland/NFTService.spec.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+            aTxHash
+          )
+        })
+
+        it("should have called send transaction with the erc721's crafted contract using the nft's chain id, the contract address and the ABI", async () => {
+          await nftService.transfer(wallet, anAddress, nft)
+          expect(walletUtils.sendTransaction as jest.Mock).toHaveBeenCalled()
+
+          const firstParameter = (walletUtils.sendTransaction as jest.Mock).mock
+            .calls[0][0] as ContractData
```
Comment (fzavalia) [inline webapp/src/modules/vendor/decentraland/NFTService.spec.ts:247]: I think there is an assert function for this called `.toHaveBeenCalledWith` that could help you not have to get into the mock data.

Comment (fzavalia) [reply webapp/src/modules/vendor/decentraland/NFTService.spec.ts:247]: You used that method further in the code 👍

File: webapp/src/modules/vendor/decentraland/NFTService.ts [hunk 2]
```diff
@@ -20,21 +20,28 @@ export class NFTService
...
-        account = this.toAccount(address)
-      }
-      account.nftIds.push(result.nft.id)
-      nfts.push({ ...result.nft, vendor: VendorName.DECENTRALAND })
-      if (result.order) {
-        orders.push(result.order)
-      }
-    }
+    const accounts: Account[] = results.reduce(
+      (accumulator: Account[], nftResult: NFTResult) => {
```
Comment (fzavalia) [inline webapp/src/modules/vendor/decentraland/NFTService.ts:24]: Types are infered by the initial accumulator and the results array, you can remove them

File: webapp/src/modules/vendor/decentraland/NFTService.spec.ts (hunk 1 above)
Comment (LautaroPetaccio) [reply webapp/src/modules/vendor/decentraland/NFTService.spec.ts:247]: `toHaveBeenCalledWith` can be used only if you know in advance both arguments. The problem with this is that the second parameter of the method is a lambda function and the only way to verify that is does what we think it does it to execute it. There's no equality comparison between function if we're not comparing references.

File: webapp/src/modules/vendor/decentraland/NFTService.ts (hunk 2 above)
Comment (LautaroPetaccio) [reply webapp/src/modules/vendor/decentraland/NFTService.ts]: You're right! I have just changed it.

File: webapp/src/modules/vendor/decentraland/NFTService.spec.ts (hunk 1 above)
Comment (fzavalia) [reply webapp/src/modules/vendor/decentraland/NFTService.spec.ts:247]: In that case you can use `expect.any(Function)` as the second argument

Comment (fzavalia) [reply webapp/src/modules/vendor/decentraland/NFTService.spec.ts:247]: Or even `expect.anything()` if you don't care for it to a function

Comment (LautaroPetaccio) [reply webapp/src/modules/vendor/decentraland/NFTService.spec.ts]: That is great! I completely forgot about those matchers.

Comment (fzavalia) [reply webapp/src/modules/vendor/decentraland/NFTService.spec.ts]: Ending with something like:

```
          expect(walletUtils.sendTransaction as jest.Mock).toHaveBeenCalledWith(
            {
              chainId: nft.chainId,
              name: 'ERC721',
              abi: ERC721Abi,
              address: nft.contractAddress,
              version: '1'
            },
            expect.any(Function)
          )
```

**Review by fzavalia — APPROVED** [review]
LGTM!
Nice job 👍

**Pull request merged**